src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mens = century
+    ? people.filter(men => Math.ceil(men.died / 100)
+  === century && (men.sex === 'm'))
+    : people.filter(person => person.sex === 'm');
+
+  const ages = mens.map(men => men.died - men.born);
+
+  const sumOfAges = ages.reduce((total, age) => {
+    return total + age;
+  }, 0);
```

Comment:
you use identical reduce three times, it is better to create a separate function with this reduce and use this function